pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -212,6 +212,11 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(proxy *model.Proxy,
 				// Only need the authentication MTLS mode when service is not external.
 				policy, _ := push.AuthenticationPolicyForWorkload(service, port)
```

Comment:
May add a TODO to remove serviceMTLSMode once alpha API is removed. We can simply use the `meshExternal` for the decision to create socket match.

Reply:
done.

Reply:
done.